### PR TITLE
BRIDGE-3228 Permissions for authorization restructure

### PR DESCRIPTION
**Pull request by jnelsonjava**
[https://sagebionetworks.jira.com/browse/BRIDGE-3230](https://sagebionetworks.jira.com/browse/BRIDGE-3230)

- Adding tests for permission CRUD operations and exception handling.

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/PermissionsTest.java [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+    private Study study;
+    
+    private String assessmentId;
+    private Assessment assessment;
+    
+    @Before
+    public void before() throws Exception {
+        admin = TestUserHelper.getSignedInAdmin();
+        user1 = TestUserHelper.createAndSignInUser(PermissionsTest.class, false, Role.DEVELOPER);
+        user2 = TestUserHelper.createAndSignInUser(PermissionsTest.class, false, Role.RESEARCHER);
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/PermissionsTest.java:64]: Don't need to have roles.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/PermissionsTest.java [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+            assertEquals("This permission cannot be created or updated because the referenced assessment does not exist.",
+                    exception.getMessage());
+        }
+        
+        
+        // Updating a permission
+        permitUser1Assessment.setAccessLevel(EDIT);
+        
+        PermissionDetail permDetUser1AssessmentUpdated = permissionsApi.updatePermission(
+                permDetUser1Assessment.getGuid(), permitUser1Assessment).execute().body();
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/PermissionsTest.java:194]: You might try breaking some or all of the other fields and verify that the server ignores this, only updating the access level of the entity

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/PermissionsTest.java [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        assertTrue(getElement(permissionsForUser1, PermissionDetail::getEntityId, assessmentId).isPresent());
+        assertTrue(getElement(permissionsForUser1, PermissionDetail::getEntityId, orgId).isPresent());
+        assertTrue(getElement(permissionsForUser1, PermissionDetail::getEntityId, studyId).isPresent());
+        
+        List<PermissionDetail> permissionsForUser2 = permissionsApi.getPermissionsForUser(user2.getUserId()).execute().body();
+        assertNotNull(permissionsForUser2);
+        assertEquals(2, permissionsForUser2.size());
+        
+        // Getting permissions for an entity
+        List<PermissionDetail> permissionsForEntity = permissionsApi.getPermissionsForEntity("PARTICIPANTS", studyId).execute().body();
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/PermissionsTest.java:252]: it drives me nuts that we can't use an enumeration here, but this is what Swagger sticks us with

**Review by alxdarksage — APPROVED** [review]
Besides the ResourceList, not much will need to change here.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/PermissionsTest.java (hunk 2 above)
Comment (jnelsonjava) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/PermissionsTest.java:194]: Added that check.

**Review by alxdarksage — APPROVED** [review]

**Comment by DwayneJengSage** [comment]
When you re-make this change, make sure to update the rest-client in pom.xml. It looks like one of the integ test failures was because Integ Tests didn't have the up-to-date JavaSDK.

**Pull request merged**